crates/circuit/src/dag_circuit.rs

```diff
@@ -7531,146 +7507,142 @@ mod test {
 
     #[test]
     fn test_push_back() -> PyResult<()> {
-        Python::with_gil(|py| {
```

Comment:
It's nice to see the Python token no longer needed in the unit tests. Probably worth noting though that there could still be issues (e.g. that import deadlock error) until we aren't using `Python::with_gil` within the DAG's methods.